### PR TITLE
Add Safari Technology Preview.app

### PR DESCRIPTION
**Pull request by jlowgren**

### CODE REVIEW

**Comment by vitorgalvao** [comment]
Leaving [on hold](https://github.com/caskroom/homebrew-cask/pull/20171#issuecomment-203700898).

File: Casks/safari-technology-preview.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  sha256 :no_check
+
+  url 'http://appldnld.apple.com/STP/SafariTechnologyPreview.dmg'
+  name 'Safari Technology Preview'
+  homepage 'https://developer.apple.com/safari/technology-preview/'
+  license :unknown
+
+  pkg 'Safari Technology Preview for OS X El Capitan.pkg'
+
+  uninstall delete: '/Applications/Safari Technology Preview.app'
```
Comment (vitorgalvao) [inline Casks/safari-technology-preview.rb]: Are you sure this is the only thing that is modified via the `pkg`?

**Comment by jlowgren** [comment]
@vitorgalvao Updated with relevant `plist`. Thanks for the heads up.

**Comment by jlowgren** [comment]
@vitorgalvao Aaaand closed it by mistake...

File: Casks/safari-technology-preview.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  sha256 :no_check
+
+  url 'https://secure-appldnld.apple.com/STP/SafariTechnologyPreview.dmg'
+  name 'Safari Technology Preview'
+  homepage 'https://developer.apple.com/safari/technology-preview/'
+  license :unknown
+
+  pkg 'Safari Technology Preview for OS X El Capitan.pkg'
+
+  uninstall pkgutil: 'com.apple.SafariTechnologyPreview.plist',
```
Comment (vitorgalvao) [inline Casks/safari-technology-preview.rb]: Isn’t this just a preferences file=

Comment (jlowgren) [inline Casks/safari-technology-preview.rb]: @vitorgalvao Updated.

File: Casks/safari-technology-preview.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+  url 'https://secure-appldnld.apple.com/STP/SafariTechnologyPreview.dmg'
+  name 'Safari Technology Preview'
+  homepage 'https://developer.apple.com/safari/technology-preview/'
+  license :unknown
+
+  pkg 'Safari Technology Preview for OS X El Capitan.pkg'
+
+  uninstall delete: [
+                      '~/Library/Preferences/com.apple.SafariTechnologyPreview.plist',
```
Comment (vitorgalvao) [inline Casks/safari-technology-preview.rb]: This being a preference, it belongs in [`zap`](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/zap.md), no `uninstall`.

But we still [haven’t decided where this cask should go](https://github.com/caskroom/homebrew-cask/pull/20171#issuecomment-203700898).

Comment (jlowgren) [inline Casks/safari-technology-preview.rb]: That's fine. I'll keep it updated until whichever outcome.

**Comment by vitorgalvao** [comment]
We’ve decided to go with this repo. Mind copying over the other improvements to the `zap` from [the other one](https://github.com/caskroom/homebrew-cask/pull/20171/files)?

File: Casks/safari-technology-preview.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+cask 'safari-technology-preview' do
+  version :latest
+  sha256 :no_check
+
+  url 'https://secure-appldnld.apple.com/STP/SafariTechnologyPreview.dmg'
+  name 'Safari Technology Preview'
+  homepage 'https://developer.apple.com/safari/technology-preview/'
+  license :unknown
```
Comment (Eitot) [inline Casks/safari-technology-preview.rb]: This should probably be `:gratis`.

Comment (vitorgalvao) [inline Casks/safari-technology-preview.rb]: Technically yes, in practice [it no longer matters](https://github.com/caskroom/homebrew-cask/issues/17437).

**Comment by zmwangx** [comment]
The files deleted under the `zap` stanza here are totally wrong. Please take a look at https://github.com/caskroom/homebrew-cask/pull/20171/files.

**Comment by adidalal** [comment]
Thank you for the contribution. It needed some fixes, so they were made in fff160a9ce6b3ff08baf245b9f8011d5b770d4fa. Your contribution is still included (and still credited to you), with the appropriate modifications. Please feel free to ask about any of the changes.

**Comment by jlowgren** [comment]
@zmwangx Gah. Copy paste fail. It was the result of

```find . -type f -iname "*technologypreview*"```

Problem is I was *in* the `~/Library` folder as I executed it. 😄

**Pull request closed without merging**